src/fprime/fbuild/builder.py

```diff
@@ -110,27 +110,29 @@ def load(self, platform: str = None, build_dir: Path = None, skip_validation=Fal
             InvalidBuildCacheException: the build cache does not exist as it must
         """
         self.__setup_default(platform, build_dir)
-        if not skip_validation and (
-            not self.build_dir.exists()
-            or not (self.build_dir / "CMakeCache.txt").exists()
-        ):
-            # Message for hard-supplied --build-cache message
-            if build_dir is not None:
-                gen_args = f" --build-cache {build_dir}"
-            else:
-                gen_args = " --ut" if self.build_type == BuildType.BUILD_TESTING else ""
-                gen_args += (
-                    " " + platform
-                    if platform is not None
-                    and platform != "native"
-                    and platform != "default"
-                    else ""
-                )
-            msg = f"'{self.build_dir}' is not a valid build cache. Generate this build cache with 'fprime-util generate{gen_args} ...'"
-            raise InvalidBuildCacheException(
-                msg,
-                self.build_dir,
+
+        if skip_validation:
+            return
+        if self.build_dir.exists() or (self.build_dir / ".fprime-build-dir").exists():
```

Comment:
This will evaluate to `True` and therefore not error if `self.build_dir` exists but `self.build_dir/.fprime-build-dir` doesn't, I'm not sure that this is what we want?

```suggestion
        if self.build_dir is not None and (self.build_dir / ".fprime-build-dir").exists():
```